File: Enigma-Machine-master/src/me/matt/enigma/wrappers/PlugBoard.java
package me.matt.enigma.wrappers;

public class PlugBoard {

    PlugBoardSwitch[] switches;

    public PlugBoard(final PlugBoardSwitch[] switches) {
        this.switches = switches;
    }

    public char swap(final char c) {
        for (final PlugBoardSwitch s : switches) {
            if (s.checkSwap(c)) {
                return s.swap(c);
            }
        }
        return c;
    }

}

File: app/src/main/java/com/sdr/enigmasimulator/enigma/PlugBoard.kt
package com.sdr.enigmasimulator.enigma


class PlugBoard(switches: Array<PlugBoardSwitch>) {
    var switches: Array<PlugBoardSwitch>
    fun swap(c: Char): Char {
        for (s in switches) {
            if (s.checkSwap(c)) {
                return s.swap(c)
            }
        }
        return c
    }

    init {
        this.switches = switches
    }
}